hal/armv7m/imxrt/timer.c

```diff
@@ -24,7 +25,47 @@
 #endif
 
 
+static struct {
+	intr_handler_t handler;
+	volatile time_t jiffies;
+	spinlock_t sp;
+
+	u32 interval;
+} timer_common;
+
+
+int _timer_irqHandler(unsigned int n, cpu_context_t *ctx, void *arg)
+{
+	(void)n;
+	(void)arg;
+	(void)ctx;
+
+	timer_common.jiffies += timer_common.interval;
+	return -1;
+}
+
+
+time_t hal_getTimer(void)
+{
+	spinlock_ctx_t sc;
+	time_t ret;
+
+	hal_spinlockSet(&timer_common.sp, &sc);
+	ret = timer_common.jiffies;
+	hal_spinlockClear(&timer_common.sp, &sc);
+
+	return ret;
+}
+
+
 void _timer_init(u32 interval)
```

Comment:
`timer_common.jiffies` is not initialized with value 0. I am not sure whether `.bss` is zeored in kernel. This applies to all platforms.

Reply:
Good find, it's not zeroed indeed